File: frontend/src/services/authService.ts
import {axiosPublic} from "../utils";

export interface LoginRequest {
    email: string;
    password: string;
}

export interface RegisterRequest {
    email: string;
    password: string;
}

const API_URL = "/api/v1/public/auth";

const register = (registerRequest: RegisterRequest) => {
    return axiosPublic.post(`${API_URL}/register`, { ...registerRequest });
};

const login = (loginRequest: LoginRequest) => {
    return axiosPublic.post(`${API_URL}/login`, { ...loginRequest });
};

const refresh = () => {
    return axiosPublic.post(`${API_URL}/refresh`);
};

const logout = () => {
    return axiosPublic.post(`${API_URL}/logout`);
};

export default {
    register,
    login,
    refresh,
    logout,
};

File: backend/src/main/java/com/sdk/itjobs/service/auth/AuthService.java
package com.sdk.itjobs.service.auth;

import com.sdk.itjobs.dto.auth.request.LoginRequest;
import com.sdk.itjobs.dto.auth.request.RegistrationRequest;
import com.sdk.itjobs.dto.auth.response.TokenResponse;
import com.sdk.itjobs.dto.user.response.UserResponse;
import com.sdk.itjobs.exception.ResourceAlreadyExistsException;
import com.sdk.itjobs.exception.ResourceNotFoundException;
import com.sdk.itjobs.exception.auth.CorruptedTokenException;
import com.sdk.itjobs.exception.auth.IncorrectPasswordException;

import org.springframework.data.util.Pair;

public interface AuthService {
    UserResponse register(RegistrationRequest request) throws ResourceAlreadyExistsException;

    Pair<TokenResponse, String> login(LoginRequest request)
            throws ResourceNotFoundException, IncorrectPasswordException;

    Pair<TokenResponse, String> refresh(String refreshToken)
            throws CorruptedTokenException, ResourceNotFoundException;

    void logout(String refreshToken);
}
